bin/docserv-createconfig

```diff
@@ -267,6 +267,21 @@ def check_langs(langs, message, attempts = 3):
     sys.exit("Expected one or more of these:\n"
         "{}".format(", ".join(languages)))
 
+def check_productid(productid, message, attempts = 3):
+    productid_regex = r'(?P<productid>[-_.+a-zA-Z0-9]+)'
+    message_expected = ("Following chars accepted: -_.+a-zA-Z0-9")
+    while attempts:
+        if not productid:
+            productid = input(message)
+        match = re.match(productid_regex, productid)
+        if match and match.group('productid') == productid:
+            return productid
+        else:
+            attempts -= 1
+            print("Invalid Product ID '{}' given".format(productid))
+            print(message_expected)
+            productid = ""
```

Comment:
It's a minor detail, but I think, the `else` is unnecessary: either you return or you proceed. As such, I'd suggest:

```python
        if match and match.group('productid') == productid:
            return productid

        attempts -= 1
        print("Invalid Product ID '{}' given".format(productid))
        print(message_expected)
        productid = ""
```

Reply:
I think you misunderstood the intent here -- this allows people three attempts to get this right before the software gives up and sys.exit(1)s.